src/overlays/actors/ovl_En_Ot/z_en_ot.c

```diff
@@ -41,128 +81,1099 @@ const ActorInit En_Ot_InitVars = {
+    EnOt* this = THIS;
+    s32 pad;
+
+    this->actionFunc(this, globalCtx);
+    Actor_SetHeight(&this->actor, 12.0f);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_80B5BB38(&this->unk_747, &this->unk_744, 0.699999988079f);
+    if (this->unk_32C & 0x400) {
+        func_80B5BAAC(&this->lightInfo, &this->unk_378, &this->unk_747, 210);
+    }
+}
+
+void func_80B5DB6C(Actor* thisx, GlobalContext* globalCtx) {
+    EnOt* this = THIS;
+    s32 pad;
+    Player* player = PLAYER;
+
+    if (!(gSaveContext.weekEventReg[84] & 0x10) && !(this->unk_32C & 8)) {
+        if (gSaveContext.weekEventReg[25] & 4) {
+            Vec3f sp50;
+
+            func_80B5B2E0(globalCtx, &this->actor.world.pos, ENOT_GET_7F(&this->actor), &sp50, &this->unk_340);
+            if (Actor_SpawnAsChildAndCutscene(&globalCtx->actorCtx, globalCtx, ACTOR_EN_OT, sp50.x, sp50.y, sp50.z, 0,
+                                              this->actor.shape.rot.y, 1, ENOT_GET_3FFF(&this->actor) | 0x8000,
+                                              this->actor.cutscene, this->actor.unk20, NULL) != NULL) {
+                this->unk_32C |= 8;
+            }
+        } else if (D_80B5E888 != NULL) {
+            s32 sp4C = false;
+
+            if (gSaveContext.weekEventReg[13] & 1) {
+                if (!func_800C9B90(&globalCtx->colCtx, player->actor.floorPoly, player->actor.floorBgId)) {
+                    sp4C = true;
+                }
+            }
+
+            if (!sp4C) {
+                EnOt* temp = D_80B5E888;
+
+                temp->unk_32C |= 8;
+                temp->unk_346 = ENOT_GET_7F(&this->actor);
+                temp->actor.params = this->actor.params;
+                temp->actor.cutscene = this->actor.cutscene;
+                this->unk_32C |= 8;
+            }
+        } else if (func_800C9B90(&globalCtx->colCtx, player->actor.floorPoly, player->actor.floorBgId)) {
+            player->unk_B2B = 29;
+        }
+    }
+}
+
+void EnOt_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnOt* this = THIS;
+    s32 pad[2];
+    Gfx* gfx;
+
+    SysMatrix_StatePush();
+    func_80B5E1D8(globalCtx, this->unk_3A4, 10);
+    SysMatrix_StatePop();
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    POLY_OPA_DISP = Gfx_CallSetupDL(POLY_OPA_DISP, 25);
+    POLY_XLU_DISP = func_8012C2B4(POLY_XLU_DISP);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+
+    AnimatedMat_Draw(globalCtx, (AnimatedMaterial*)Lib_SegmentedToVirtual(&D_060005F8));
+    SkelAnime_DrawSV(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, this->skelAnime.dListCount, NULL,
+                     EnOt_PostLimbDraw, &this->actor);
+    SysMatrix_InsertTranslation(this->unk_378.x, this->unk_378.y, this->unk_378.z, MTXMODE_NEW);
+    Matrix_Scale(0.0882f, 0.0882f, 0.0882f, MTXMODE_APPLY);
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    gfx = func_8012C7FC(POLY_XLU_DISP);
+
+    gSPSetOtherMode(&gfx[0], G_SETOTHERMODE_H, 4, 4, 0x00000080);
```

Comment:
I suspect there's a macro for that last number.

Reply:
I just copied what Tharo said here https://discord.com/channels/688807550715560050/688851997352263820/858107969681883156